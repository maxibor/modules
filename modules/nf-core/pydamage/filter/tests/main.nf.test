
nextflow_process {

    name "Test Process PYDAMAGE_FILTER"
    script "../main.nf"
    process "PYDAMAGE_FILTER"

    tag "modules"
    tag "modules_nfcore"
    tag "pydamage"
    tag "pydamage/filter"

    test("test-pydamage-filter") {

        setup {
            run ("PYDAMAGE_ANALYZE") {
                script "../../../pydamage/analyze/main.nf"
                process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                ]
                """
                }
            }
        }

        when {
            process {
                """
                input[0] = PYDAMAGE_ANALYZE.out.csv
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
