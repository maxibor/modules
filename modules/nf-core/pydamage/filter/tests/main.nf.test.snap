{
    "test-pydamage-filter": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_pydamage_filtered_results.csv:md5,6414a400229f9cc5e31368703020b039"
                    ]
                ],
                "1": [
                    "versions.yml:md5,6c9e5b134c5734791bbf1d857be251c9"
                ],
                "csv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_pydamage_filtered_results.csv:md5,6414a400229f9cc5e31368703020b039"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6c9e5b134c5734791bbf1d857be251c9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-25T13:26:34.666732"
    }
}